### PR TITLE
Implement signal for core plugins + update docs

### LINKED ISSUE #548: Signal resources
Many resources have state transitions that would be useful to be able to trigger easily, like start/stop/restart a VM or service.

Add a new Signal primitive that plugin entries can implement to allow sending arbitrary signals to a resource. Signal names are case-insensitive.
```
type Signalable interface {
  Signal(context.Context, string) error
}
```

We would like to establish some common conventions for this that make it easier to script without overly constraining the possible applications. As such, add to the plugin documentation around Signalable that `start`, `stop`, and `restart` are common signal names that you should re-use if possible.

Additionally `EntrySchema` should be updated to include a map of supported signals and descriptions of how the entry will respond to them. Those descriptions should be accessible through the `docs` command.

The Wash API should respond to `POST /fs/signal?path=<path>` by invoking Signal on the specified entry and returning whether sending the signal succeeded.

Add a new `signal` command that accepts `signal <signal name> path/to/resource`. It should error and display supported signals if the supplied signal name doesn't match one accepted by the entry at the specified path. It should return after sending the signal; waiting until the signaled operation has completed is an interesting concept, but not one we can do generally for any signal.

See https://docs.google.com/document/d/1NnJ1Qxeu7Sh4_ymOpo0enibKb3V5tzsSsX3H-N0PPDY/edit?usp=sharing for further discussion on the concept.

### PR DESCRIPTION
**Pull request by ekinanp**
Resolves https://github.com/puppetlabs/wash/issues/548

### CODE REVIEW

File: plugin/aws/ec2Instance.go [hunk 1]
```diff
@@ -159,7 +159,12 @@ func (inst *ec2Instance) Schema() *plugin.EntrySchema {
 	return plugin.
 		NewEntrySchema(inst, "instance").
 		SetDescription(ec2InstanceDescription).
-		SetMetaAttributeSchema(ec2InstanceMetadata{})
+		SetMetaAttributeSchema(ec2InstanceMetadata{}).
+		AddSignal("start", "Starts the EC2 instance").
```
Comment (ekinanp) [inline plugin/aws/ec2Instance.go]: Think I'm going to refactor wash-ruby to adopt similar syntax.

Comment (ekinanp) [reply plugin/aws/ec2Instance.go]: https://github.com/puppetlabs/wash-ruby/pull/20

File: docs/docs/index.md [hunk 2]
```diff
@@ -136,7 +140,15 @@ Each line represents validation of an entry type. The `lrsx` fields represent su
...
-Displays the entry's documentation. This is currently its description.
+Displays the entry's documentation. This is currently its description and any supported signals/signal groups.
+
+### wash delete
+
+Deletes the entries at the specified paths, prompting the user for confirmation before deleting each entry.
+
+### wash signal
+
+Sends the specified signal to the entry at the specified path
```
Comment (MikaelSmith) [inline docs/docs/index.md]: Should end with a period.

File: plugin/aws/ec2Instance.go [hunk 3]
```diff
@@ -255,10 +260,7 @@ func (inst *ec2Instance) checkLatestConsoleOutput(ctx context.Context) (*ec2Inst
 }
 
 func (inst *ec2Instance) Delete(ctx context.Context) (bool, error) {
-	_, err := inst.client.TerminateInstancesWithContext(ctx, &ec2Client.TerminateInstancesInput{
-		InstanceIds: awsSDK.StringSlice([]string{inst.id}),
-	})
-	return false, err
+	return false, inst.Signal(ctx, "terminate")
```
Comment (MikaelSmith) [inline plugin/aws/ec2Instance.go]: It's weird that there's two ways to do this. But not necessarily a bad idea.

Comment (ekinanp) [reply plugin/aws/ec2Instance.go]: Yeah, opted for keeping things symmetrical w/ AWS docs

File: docs/docs/index.md (hunk 2 above)
Comment (ekinanp) [reply docs/docs/index.md]: Thanks, updated.

**Comment by ekinanp** [comment]
Might be interesting to have something like `signal(ctx, signal, opts)` where `opts` is JSON specifying the signal options. SignalSchema can be updated to include these options that we can check during validation. That would let us have access to all options for a given API request.

We can include the options for a given signal/signal group as part of docs' output.

Am OK with deferring that till it's needed though.

**Comment by MikaelSmith** [comment]
Yes we should defer additional signal options. I'd want a little time to hash out the CLI options for it and some signal that it's needed.

**Review by MikaelSmith — APPROVED** [review]

**Pull request merged**